cert-manager/controllers/keptnwebhookcontroller/keptnwebhookcertificate_controller.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keptnwebhookcontroller
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/keptn/lifecycle-toolkit/cert-manager/eventfilter"
+	"github.com/pkg/errors"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	SuccessDuration              = 3 * time.Hour
+	Webhookconfig                = "klc-mutating-webhook-configuration"
+	secretPostfix                = "-certs"
+	errorCertificatesSecretEmpty = "certificates secret is empty"
+	ns                           = "keptn-lifecycle-toolkit-system"
+)
+
+// KeptnWebhookCertificateReconciler reconciles a KeptnWebhookCertificate object
+type KeptnWebhookCertificateReconciler struct {
+	ctx           context.Context
+	Client        client.Client
+	Scheme        *runtime.Scheme
+	ApiReader     client.Reader
+	CancelMgrFunc context.CancelFunc
+	Log           logr.Logger
+}
+
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,namespace=keptn-lifecycle-toolkit-system,resourceNames=klc-mutating-webhook-configuration,resources=mutatingwebhookconfigurations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",namespace=keptn-lifecycle-toolkit-system,resources=secrets,verbs=get;list;watch;create;update;patch;delete
```

Comment:
cert manager need bot read and write access to secrets in our namespace since it checks if the secret is valid and updates it